PKHeX.WinForms/Subforms/Save Editors/Gen3/SAV_Misc3.cs

```diff
@@ -187,5 +195,188 @@ private void saveFerry()
             setFerryFlagFromNum(0x1DB, CHK_InitialNavel.Checked);
         }
         #endregion
+
+        #region BattleFrontier
+        private int[] Symbols;
+        private int ofsSymbols;
+        private System.Drawing.Color[] SymbolColorA;
+        private Button[] SymbolButtonA;
+        private bool editingcont;
+        private bool editingval;
+        private RadioButton[] StatRBA;
+        private NumericUpDown[] StatNUDA;
+        private Label[] StatLabelA;
+        private bool loading;
+        private int[][] BFF;
+        private string[][] BFT;
+        private int[][] BFV;
+        private string[] BFN;
+        private void ChangeStat1(object sender, EventArgs e)
+        {
+            if (loading) return;
+            int facility = CB_Stats1.SelectedIndex;
+            if (facility < 0 || facility >= BFN.Length) return;
+            editingcont = true;
+            CB_Stats2.Items.Clear();
+            foreach (RadioButton r in StatRBA)
+                r.Checked = false;
+
+            if (BFT[BFF[facility][1]] == null) CB_Stats2.Visible = false;
+            else
+            {
+                CB_Stats2.Visible = true;
+                for (int i = 0; i < BFT[BFF[facility][1]].Length; i++)
+                    CB_Stats2.Items.Add(BFT[BFF[facility][1]][i]);
+                CB_Stats2.SelectedIndex = 0;
+            }
+
+            for (int i = 0; i < StatLabelA.Length; i++)
+                StatLabelA[i].Visible = StatLabelA[i].Enabled = StatNUDA[i].Visible = StatNUDA[i].Enabled = Array.IndexOf(BFV[BFF[facility][0]], i) >= 0;
+
+            editingcont = false;
+            StatRBA[0].Checked = true;
+        }
+        private void ChangeStat(object sender, EventArgs e)
+        {
+            if (editingcont) return;
+            StatAddrControl(SetValToSav: -2, SetSavToVal: true);
+        }
+        private void StatAddrControl(int SetValToSav = -2, bool SetSavToVal = false)
+        {
+            int Facility = CB_Stats1.SelectedIndex;
+            if (Facility < 0) return;
+
+            int BattleType = CB_Stats2.SelectedIndex;
+            if (BFT[BFF[Facility][1]] == null) BattleType = 0;
+            else if (BattleType < 0) return;
+            else if (BattleType >= BFT[BFF[Facility][1]].Length) return;
+
+            int RBi = -1;
+            for (int i = 0, j = 0; i < StatRBA.Length; i++)
+            {
+                if (!StatRBA[i].Checked) continue;
+                if (++j > 1) return;
+                RBi = i;
+            }
+            if (RBi < 0) return;
```

Comment:
This loop could be eliminated by using Linq. Something like this:
RBi = StatRBF.FirstOrDefault(x => x.Checked) ?? -1;